### PR TITLE
Cursors are using the closest point instead of linear interpolation

### PR DESCRIPTION
**Pull request by ralucaportase**

### CODE REVIEW

**Comment by ralucaportase** [comment]
The tests are failing here because it does not see the api from jquery.flot

File: jquery.flot.cursors.js [hunk 1]
```diff
@@ -334,49 +334,27 @@ Licensed under the MIT license.
...
-                    y = p2y
-                } else {
-                    y = p1y + (p2y - p1y) * (pos.x - p1x) / (p2x - p1x);
-                }
-
-                pos.y = y;
-
+            
+            var appropiatePoint = plot.findNearbyItem(cursor.x, cursor.y , function(i) {
+                        return i === cursor.snapToPlot;
```
Comment (cristiingineru) [inline jquery.flot.cursors.js]: Renaming the i argument to maybe seriesIndex would increase the readability of this function.

File: jquery.flot.cursors.js [hunk 2]
```diff
@@ -334,49 +334,27 @@ Licensed under the MIT license.
...
+                    }, Number.MAX_VALUE);
+                        
+            if(appropiatePoint!=null){
+                var dataset = plot.getData(),
+                    points = dataset[appropiatePoint.seriesIndex].datapoints.points,
+                    ps = dataset[appropiatePoint.seriesIndex].datapoints.pointsize,
+                    i = appropiatePoint.dataIndex * ps;
+                    
+                pos.y = appropiatePoint.datapoint[1];
+                pos.x = appropiatePoint.datapoint[0];
```
Comment (cristiingineru) [inline jquery.flot.cursors.js]: It doesn't look like pos.y and pos.x are red back, so they should be deleted.

File: jquery.flot.cursors.js [hunk 3]
```diff
@@ -334,49 +334,27 @@ Licensed under the MIT license.
...
-                }
-
-                pos.y = y;
-
+            
+            var appropiatePoint = plot.findNearbyItem(cursor.x, cursor.y , function(i) {
+                        return i === cursor.snapToPlot;
+                    }, Number.MAX_VALUE);
+                        
+            if(appropiatePoint!=null){
```
Comment (cristiingineru) [inline jquery.flot.cursors.js]: In JS it's fine to do the null checks with:
if(appropiatePoint) {
   ...
}

File: jquery.flot.cursors.js [hunk 4]
```diff
@@ -334,49 +334,27 @@ Licensed under the MIT license.
...
+                var dataset = plot.getData(),
+                    points = dataset[appropiatePoint.seriesIndex].datapoints.points,
+                    ps = dataset[appropiatePoint.seriesIndex].datapoints.pointsize,
```
Comment (cristiingineru) [inline jquery.flot.cursors.js]: Is appropiatePoint.seriesIndex the same with cursor.snapToPlot?

Comment (ralucaportase) [inline jquery.flot.cursors.js]: I think so

File: jquery.flot.cursors.js [hunk 5]
```diff
@@ -334,49 +334,25 @@ Licensed under the MIT license.
...
-                if (p1x === p2x) {
-                    y = p2y
-                } else {
-                    y = p1y + (p2y - p1y) * (pos.x - p1x) / (p2x - p1x);
-                }
-
-                pos.y = y;
-
+            
+            var appropiatePoint = plot.findNearbyItem(cursor.x, cursor.y , function(seriesIndex) {
```
Comment (cipix2000) [inline jquery.flot.cursors.js]: closestPoint or nearestPoint sounds better to me

File: spec/cursors_intersections.Test.js [hunk 6]
```diff
@@ -78,15 +80,16 @@ describe("Cursors intersections", function () {
     });
 
 
-    it('should find intersections with multiple plots', function () {
+    xit('should find intersections with multiple plots', function () {
```
Comment (cipix2000) [inline spec/cursors_intersections.Test.js]: remove the tests that don't make sense anymore

Comment (cipix2000) [inline spec/cursors_intersections.Test.js]: remove also 
  it('should interpolate the intersections properly with log scales');
  it('should interpolate the intersections properly with mixed scales');

**Pull request merged**